### PR TITLE
Add iommu support for all virtio devices

### PR DESCRIPTION
**Pull request by lijinlijin**
Add iommu support for all virtio devices
ID: 1783824
Signed-off-by: Li Jin <lijin@redhat.com>

### CODE REVIEW

**Comment by lijinlijin** [comment]
(1/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_keyboard.q35: PASS (125.56 s)
 (2/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_keyboard.iommu_enabled.q35: PASS (122.30 s)
 (3/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_mice.device_mouse.q35: PASS (97.10 s)
 (4/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_mice.device_tablet.q35: PASS (98.33 s)
 (5/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_mice.iommu_enabled.device_mouse.q35: PASS (113.51 s)
 (6/6) Host_RHEL.m8.u1.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.vioinput_mice.iommu_enabled.device_tablet.q35: PASS (112.42 s)

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -2351,6 +2351,8 @@ def sort_key(dev):
         virtio_pci_devices = ["virtio-net-pci", "virtio-blk-pci",
                               "virtio-scsi-pci", "virtio-balloon-pci",
                               "virtio-serial-pci", "virtio-rng-pci"]
+                              "virtio-keyboard-pci", "virtio-mouse-pci",
```
Comment (peixiu) [inline virttest/qemu_vm.py]: repeated ']' in line 2353, need to change to ','

Comment (lijinlijin) [reply virttest/qemu_vm.py]: Yes, modified, thanks

**Comment by yanan-fu** [comment]
May be we can refer this:  https://github.com/avocado-framework/avocado-vt/pull/2463
Make it easier to maintain.

**Comment by lijinlijin** [comment]
@yanan-fu Modified, thanks

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -2348,9 +2348,6 @@ def sort_key(dev):
             add_qemu_option(devices, "k", [attr_info])
 
         # Add disable_legacy and disable_modern options
-        virtio_pci_devices = ["virtio-net-pci", "virtio-blk-pci",
-                              "virtio-scsi-pci", "virtio-balloon-pci",
-                              "virtio-serial-pci", "virtio-rng-pci"]
         for device in devices:
             dev_type = device.get_param("driver")
```
Comment (xutian) [inline virttest/qemu_vm.py]: Ok, let's speed uo this code:
```
virtio_devices = filter(lambda x:x.get_param('driver','').startswith('virtio-'), devices)
for device in virtio_devices:
    disable_legacy = params.get("virtio_dev_disable_legacy", None)
    disable_modern = params.get("virtio_dev_disable_modern", None)
    iommu_platform = params.get("virtio_dev_iommu_platform", None)
 ```

**Comment by yanan-fu** [comment]
It is better.  Ack

**Comment by xutian** [comment]
@lijinlijin please import filter function in line#7 to ensure it works for all version's python.

**Pull request merged**